### PR TITLE
Analytics.config returns nil when not configured

### LINKED ISSUE #2156: Analytics errors if analytics.yml is not configured
```
> Sufia::Analytics.profile
NoMethodError: undefined method `[]' for false:FalseClass
	from /opt/rubies/ruby-2.3.0/lib/ruby/gems/2.3.0/gems/sufia-7.0.0.beta1/app/services/sufia/analytics.rb:16:in `config'
	from /opt/rubies/ruby-2.3.0/lib/ruby/gems/2.3.0/gems/sufia-7.0.0.beta1/app/services/sufia/analytics.rb:23:in `token'
	from /opt/rubies/ruby-2.3.0/lib/ruby/gems/2.3.0/gems/sufia-7.0.0.beta1/app/services/sufia/analytics.rb:38:in `user'
	from /opt/rubies/ruby-2.3.0/lib/ruby/gems/2.3.0/gems/sufia-7.0.0.beta1/app/services/sufia/analytics.rb:44:in `profile'
```

### PR DESCRIPTION
**Pull request by jcoyne**
Fix error when analytics.yml is not configured.
Instead log an error and return nil

Fixes #2156

### CODE REVIEW

**Comment by awead** [comment]
Since these are all class methods, why not:
``` ruby
class << self
  # public class methods
  private
    # private_class_methods
end
```
and then you don't need the `private_class_method.

**Comment by jcoyne** [comment]
Are they still class methods if they are in a Module?

**Comment by awead** [comment]
@jcoyne hmm. Good question. meh... something to save for later 😄

File: app/services/sufia/analytics.rb [hunk 1]
```diff
@@ -13,30 +13,47 @@ module Analytics
...
+        Rails.logger.error("Unable to fetch any keys from #{filename}.")
+        return
+      end
+      yaml.fetch('analytics')
+    end
+    private_class_method :load_config
 
     # Generate an OAuth2 token for Google Analytics
-    # @return [OAuth2::AccessToken] An OAuth2 access token for GA
+    # @return [OAuth2::ccessToken] An OAuth2 access token for GA
```
Comment (mjgiarlo) [inline app/services/sufia/analytics.rb]: Typo.

**Comment by mjgiarlo** [comment]
Happy to merge after the new YARD typo is fixed.

**Comment by jcoyne** [comment]
@mjgiarlo Typo fixed.

**Pull request merged**